### PR TITLE
#162694397 Fix slug, articles refactoring and Heroku deployment

### PR DESCRIPTION
**Pull request by GransonO**
#### What does this PR do?
This PR refactors the articles classes and functions, reconfigures Heroku deployement and slugs generation. 

#### Description of Task to be completed?
- Refactor Articles code to follow best practices
- create a better slug generator
- Reconfigure Heroku deployment

#### How should this be manually tested?
```
1. Access the swagger documentation from : [Heroku link to site](https://ah-jumanji-staging.herokuapp.com/)
2. Register a user and login
3. Authorize the user to view the rest of the endpoints
```
#### Any background context you want to provide?
Following best programming standards is important as it enebles the rest of the team members to easily onboard themselves to your code and makes learning easier. it also improves the debugging capabilities of the programmer.

#### What are the relevant pivotal tracker stories?
#[162694397](https://www.pivotaltracker.com/n/projects/2225604)
#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

### CODE REVIEW

**Review by mashafrancis — APPROVED** [review]
Awesome work with the bug fix.

**Review by fabzer0 — APPROVED** [review]

**Comment by fabzer0** [comment]
This is work is really great. I can see the slug is okay, plus  I get the author of an article without passing in the author name on request!

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -1,145 +1,39 @@
...
-            return ("Article does not exist")
-
-
-class AuthorSerializer(serializers.ModelSerializer):
+        fields = "__all__"
+
+    def author_details(self, obj):
+        user = self.context["user"]
+
+        # logger.error(user_context)
```
Comment (ghost) [inline authors/apps/articles/serializers.py]: Remove code that has been commented out

**Review by ghost — APPROVED** [review]
Neat work

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (GransonO) [reply authors/apps/articles/serializers.py]: Done

**Review by gwako94 — APPROVED** [review]
Good work with the bug fix.

**Review by EmmanuelBeja — APPROVED** [review]
Looks pretty good to me.

**Review by gitau-kinuthia — APPROVED** [review]

**Pull request merged**